examples/rpc-kitchen-sink/index.rsh

```diff
@@ -0,0 +1,29 @@
+'reach 0.1';
+
+import * as runView from './runView.rsh';
+
+export const five = 5;
+export const a    = { b: { c: 'd', e: [ five, 6 ] }};
+
+export const timesTwoPlusThree = is((x => x * 2 + 3), Fun([ UInt ], UInt));
+
+export const main = Reach.App(() => {
+  const A = Participant('Alice', {
+    ...runView.pii,
+  });
+
+  const [ v1, v2 ] = [ View('Main', runView.T), View(runView.T) ];
+
+  setOptions({ verifyArithmetic: true, verifyPerConnector: true });
+  deploy(); A.publish(); commit();
+
+  // Temporarily disable until ETH gas estimation + arbitrary view-checking order are fixed
```

Comment:
What does this mean?

Reply:
The gas part is referring to the ETH CI failure I mentioned:
https://77248-205868997-gh.circle-artifacts.com/0/tmp/artifacts/ETH.rpc-kitchen-sink.gz

The arbitrary order part is what I was saying about reshuffling the order in which `runView.vbytes` and `runView.fun` are executed. I feel like I should be able to run them in any order but I must be missing something to allow that to succeed.

Reply:
First, that error means that the program is too big. There's a maximum size of ETH programs and currently our view code is not as efficient as it could be.

Second, what error do you get when you switch the order?

Reply:
(Using `ALGO`)

If I move the `runView.fun` lines from 24 and 25 above the `runView.vbytes` lines I get:
```
backend.getExports > app.timesTwoPlusThree(app.five): {"type":"BigNumber","hex":"0x0d"} === {"type":"BigNumber","hex":"0x0d"}
backend.getExports > app.a.b.e: [{"type":"BigNumber","hex":"0x05"},{"type":"BigNumber","hex":"0x06"}] === [{"type":"BigNumber","hex":"0x05"},{"type":"BigNumber","hex":"0x06"}]
backend.getExports > app.a.b.c: "d" === "d"
backend.getExports > app.a.z.c: null === null
backend.getExports > app.nonExistent: null === null
backend.getExports > app.nonExistent.a: null === null
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: *** Mismatch! ***
  Expected: ["Some",true]
    Actual: ["None",null]
```

If I run just the `runView.fun` lines and comment both `runView.vbytes` lines it succeeds:
```
backend.getExports > app.timesTwoPlusThree(app.five): {"type":"BigNumber","hex":"0x0d"} === {"type":"BigNumber","hex":"0x0d"}
backend.getExports > app.a.b.e: [{"type":"BigNumber","hex":"0x05"},{"type":"BigNumber","hex":"0x06"}] === [{"type":"BigNumber","hex":"0x05"},{"type":"BigNumber","hex":"0x06"}]
backend.getExports > app.a.b.c: "d" === "d"
backend.getExports > app.a.z.c: null === null
backend.getExports > app.nonExistent: null === null
backend.getExports > app.nonExistent.a: null === null
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["Some",true] === ["Some",true]
ctc.views > Main/fun: ["Some",true] === ["Some",true]
ctc.v     > Main/fun: ["Some",false] === ["Some",false]
ctc.views > Main/fun: ["Some",false] === ["Some",false]
ctc.v     > Main/fun: ["Some",false] === ["Some",false]
ctc.views > Main/fun: ["Some",false] === ["Some",false]
ctc.v     > Main/fun: ["Some",false] === ["Some",false]
ctc.views > Main/fun: ["Some",false] === ["Some",false]
ctc.v     > Main/fun: ["Some",true] === ["Some",true]
ctc.views > Main/fun: ["Some",true] === ["Some",true]
ctc.v     > Main/fun: ["Some",true] === ["Some",true]
ctc.views > Main/fun: ["Some",true] === ["Some",true]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > Main/fun: ["None",null] === ["None",null]
ctc.views > Main/fun: ["None",null] === ["None",null]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
ctc.v     > fun: ["Some",true] === ["Some",true]
ctc.views > fun: ["Some",true] === ["Some",true]
ctc.v     > fun: ["Some",false] === ["Some",false]
ctc.views > fun: ["Some",false] === ["Some",false]
ctc.v     > fun: ["Some",false] === ["Some",false]
ctc.views > fun: ["Some",false] === ["Some",false]
ctc.v     > fun: ["Some",false] === ["Some",false]
ctc.views > fun: ["Some",false] === ["Some",false]
ctc.v     > fun: ["Some",true] === ["Some",true]
ctc.views > fun: ["Some",true] === ["Some",true]
ctc.v     > fun: ["Some",true] === ["Some",true]
ctc.views > fun: ["Some",true] === ["Some",true]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
ctc.v     > fun: ["None",null] === ["None",null]
ctc.views > fun: ["None",null] === ["None",null]
```

So this demonstrates what I was talking about with "subsequent phases" on Slack: I'm surprised the equality assertion would fail where it did (if it even fails at all).

It makes even less sense to me when I put `runView.fun` first and disable lines _inside_ of `runView.vbytes` instead of commenting it out entirely because that causes the assertion to fail elsewhere, but still in `runView.fun`. I also spent a good amount of time considering whether I'd messed up my consensus steps somehow but didn't make much progress on that front.

It feels like the broader takeaway is I need to crunch this down into much smaller bites, though (even if I'm just confused about how to make the above bits work)... We probably want separate contracts running under one test umbrella, eh?

Reply:
First, your last comment, yes, I think it will be simpler to make a bunch of small applications. They can all be in the same file and Reach will compile them all and your single frontend (and single RPC server) will use them all.

Second, the program is too abstract for me to track what is really happening. I don't understand what the two `runView` functions do nor what the `f` function inside them does. That said, I think that you could label the `f` calls so you'd know which ones are failing. The next thing that is odd to me is that each `runView` functions starts "unsynchronized", because there is a bare `f` call without knowing what the state is, so you don't know it will be `None`.